plugins/inputs/postgresql/postgresql.go

```diff
@@ -42,6 +40,20 @@ var sampleConfig = `
   ##
   address = "host=localhost user=postgres sslmode=disable"
 
+  ## connection configuration.
+
+  ## maxidle - specify the number of idle connections to maintain. see sql.DB for details.
+  ## default is 1. setting to 0 to force disconnect after use.
+  max_idle = 1
```

Comment:
>every single connection is going to be idle.

of which we only have one because of how the gather step is laid out. it only ever creates a single connection in its current form.


>every single connection is going to be idle.

yup that single connection is going to be idle which is fine because the default idle allowed is 1 with a forever lifetime.

>Thus it's going to result in connections constantly being created and destroyed

flat out incorrect.